wagtail_newsletter/actions.py

```diff
@@ -1,24 +1,25 @@
 from typing import cast
 
 from wagtail.admin import messages
-
-from wagtail_newsletter.forms import SendTestEmailForm
+from wagtail.log_actions import get_active_log_context
 
 from . import campaign_backends
-from .models import NewsletterPageMixin
+from .forms import SendTestEmailForm
+from .models import NewsletterLogEntry, NewsletterPageMixin
 
 
 def save_campaign(request, page: NewsletterPageMixin) -> None:
     backend = campaign_backends.get_backend()
-    revision = cast(NewsletterPageMixin, page.latest_revision.as_object())
-    subject = revision.newsletter_subject or revision.title
+    revision = page.latest_revision
+    version = cast(NewsletterPageMixin, revision.as_object())
```

Comment:
The old `revision` is now `version` because we also need the plain revision object (i.e. before calling `.as_objecet()`) to log it.